src/TaskModel.vala

```diff
@@ -239,6 +239,32 @@ public class Tasks.TaskModel : Object {
 
                 registry.refresh_backend_sync (collection_source.uid, null);
                 break;
+            case "google":
+                var collection_source = registry.find_extension (task_list, E.SOURCE_EXTENSION_COLLECTION);
+                var authorizer = (GData.Authorizer) new E.GDataOAuth2Authorizer (collection_source, typeof (GData.TasksService));
+                var service = new GData.TasksService (authorizer);
+                var resource_extension = (E.SourceResource) task_list.get_extension (E.SOURCE_EXTENSION_RESOURCE);
+
+                GData.TasksTasklist? tasklist = null;
+
+                unowned GLib.List<GData.Entry> gtasklists = service.query_all_tasklists (null, null, null).get_entries ();
+                foreach (unowned GData.Entry gtasklist in gtasklists) {
+                    if ("gtasks::%s".printf (gtasklist.id) == resource_extension.identity) {
+                        tasklist = (GData.TasksTasklist) gtasklist;
+                        break;
+                    }
+                }
```

Comment:
Ideally, shouldn't this be solved by loading the `GData.TasksTasklist` directly with something like the following?

```vala
GData.TasksTasklist? tasklist = (GData.TasksTasklist?) service.query_single_entry (
    GData.TasksService.get_primary_authorization_domain (),
    resource_extension.dup_identity (),
    null,
    typeof (GData.TasksTasklist),
    null
);
```

However, this seems to throw the following error. Is this a bug in libgdata which should be fixed?

```bash
libgdata:ERROR:../gdata/gdata-service.c:588:_gdata_service_build_message: assertion failed (soup_uri_get_scheme (_uri) == SOUP_URI_SCHEME_HTTPS): ("gtasks" == "https")
Bail out! libgdata:ERROR:../gdata/gdata-service.c:588:_gdata_service_build_message: assertion failed (soup_uri_get_scheme (_uri) == SOUP_URI_SCHEME_HTTPS): ("gtasks" == "https")
Aborted
```

Reply:
I did it first and ended on the same error, so i stick with the way we do to rename the tasklist.
But looking how libgdata use it internaly, i don't think it's a bug, we only need to pass the full method uri instead, so:
``` vala
GData.TasksTasklist tasklist = (GData.TasksTaskList) service.query_single_entry (
    GData.TasksService.get_primary_authorization_domain (),
    "https://www.googleapis.com/tasks/v1/users/@me/lists/%s".printf (resource_extension.identity.replace ("gtasks::", "")),
    null,
    typeof (GData.TasksTaskList),
    null
);
```